File: backend/util/file.go
package util

import (
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
	"regexp"
	"sort"
	"strconv"
	"strings"
)

func ReadLinesFromDirectory(dir string) ([]string, error) {
	files, err := ioutil.ReadDir(dir)
	if err != nil {
		return nil, err
	}
	// 按照数字顺序从小到大读取
	// 正则表达式用于提取文件名中的数字
	re := regexp.MustCompile(`\d+`)

	// 创建一个切片来存储文件名和对应的数字
	type fileWithNumber struct {
		name   string
		number int
	}

	var fileList []fileWithNumber

	for _, file := range files {
		if !file.IsDir() && strings.HasSuffix(file.Name(), ".txt") {
			matches := re.FindStringSubmatch(file.Name())
			if len(matches) > 0 {
				number, err := strconv.Atoi(matches[0])
				if err == nil {
					fileList = append(fileList, fileWithNumber{name: file.Name(), number: number})
				}
			}
		}
	}

	sort.Slice(fileList, func(i, j int) bool {
		return fileList[i].number < fileList[j].number
	})

	var lines []string
	for _, file := range fileList {
		content, err := os.ReadFile(filepath.Join(dir, file.name))
		if err != nil {
			// 打印错误并继续处理其他文件
			fmt.Printf("Error reading file %s: %v\n", file.name, err)
			continue
		}
		lines = append(lines, string(content))
	}
	return lines, nil
}

func ReadFilesFromDirectory(dir string) ([]os.DirEntry, error) {
	files, err := os.ReadDir(dir)
	if err != nil {
		return nil, err
	}
	// Create a slice to hold the files along with their extracted numbers
	type fileWithNumber struct {
		entry  os.DirEntry
		number int
	}
	var fileList []fileWithNumber
	for _, file := range files {
		if !file.IsDir() {
			name := file.Name()
			// Extract the numeric part from the file name
			numberStr := strings.TrimSuffix(name, filepath.Ext(name))
			number, err := strconv.Atoi(numberStr)
			if err != nil {
				fmt.Printf("failed to convert %s to number: %s\n", numberStr, err)
				continue
			}
			fileList = append(fileList, fileWithNumber{entry: file, number: number})
		}
	}
	// Sort the files based on the extracted number
	sort.Slice(fileList, func(i, j int) bool {
		return fileList[i].number < fileList[j].number
	})
	// Extract the sorted os.DirEntry from the sorted fileWithNumber slice
	sortedFiles := make([]os.DirEntry, len(fileList))
	for i, file := range fileList {
		sortedFiles[i] = file.entry
	}
	return sortedFiles, nil
}

File: backend/util/file.py
import json
import logging
import os
import re
import shutil
from idlelib.iomenu import encoding
from typing import List, Tuple

from backend.util.constant import config_path

def read_lines_from_directory_utf8(directory):
    if not os.path.exists(directory):
        logging.info(f"dir {directory} doesn't exist")
        return None, None
    try:
        files = os.listdir(directory)
    except OSError as e:
        print(f"Error reading directory {directory}: {e}")
        return None, e

    # Regular expression to extract numbers from filenames
    regex = re.compile(r'\d+')

    # List to store filenames and their corresponding numbers
    file_list = []

    for file in files:
        if file.endswith('.txt'):
            matches = regex.findall(file)
            if matches:
                try:
                    number = int(matches[0])
                    file_list.append((file, number))
                except ValueError:
                    continue

    # Sort files based on the extracted number
    file_list.sort(key=lambda x: x[1])

    lines = []
    for file, _ in file_list:
        try:
            with open(os.path.join(directory, file), 'r', encoding='gbk', errors='ignore') as f:
                lines.append(f.read())
        except OSError as e:
            print(f"Error reading file {file}: {e}")
            continue

    return lines, None

def read_lines_from_directory(directory):
    if not os.path.exists(directory):
        logging.info(f"dir {directory} doesn't exist")
        return None, None
    try:
        files = os.listdir(directory)
    except OSError as e:
        print(f"Error reading directory {directory}: {e}")
        return None, e

    # Regular expression to extract numbers from filenames
    regex = re.compile(r'\d+')

    # List to store filenames and their corresponding numbers
    file_list = []

    for file in files:
        if file.endswith('.txt'):
            matches = regex.findall(file)
            if matches:
                try:
                    number = int(matches[0])
                    file_list.append((file, number))
                except ValueError:
                    continue

    # Sort files based on the extracted number
    file_list.sort(key=lambda x: x[1])

    lines = []
    for file, _ in file_list:
        try:
            with open(os.path.join(directory, file), 'r', errors = 'ignore') as f:
                lines.append(f.read())
        except OSError as e:
            print(f"Error reading file {file}: {e}")
            continue

    return lines, None

def read_files_from_directory(dir_path: str) -> List[os.DirEntry]:
    """
    Reads all files from the specified directory, sorts them by numeric order
    extracted from their filenames, and returns them as a list of os.DirEntry objects.
    
    :param dir_path: Path to the directory containing the files.
    :return: List of os.DirEntry objects sorted by numeric order.
    """
    # Regular expression to extract numbers from filenames
    if not os.path.exists(dir_path):
        return []
    number_re = re.compile(r'\d+')
    
    # List to store tuples of (os.DirEntry, number)
    file_list = []
    
    # Iterate over directory entries
    with os.scandir(dir_path) as entries:
        for entry in entries:
            if entry.is_file():
                # Extract number from filename
                match = number_re.search(entry.name)
                if match:
                    number = int(match.group())
                    file_list.append((entry, number))
    
    # Sort files by the extracted number
    file_list.sort(key=lambda x: x[1])
    
    # Extract sorted os.DirEntry objects
    sorted_files = [entry.name for entry, _ in file_list]
    
    return sorted_files

def save_list_to_files(input_list, path, offset):
    try:
        for i, line in enumerate(input_list):
            file_path = os.path.join(path, f"{i + offset}.txt")
            with open(file_path, 'w') as file:
                file.write(line)
    except Exception as e:
        return e
    return None

def remove_all(directory):
    if os.path.exists(directory):
        shutil.rmtree(directory, ignore_errors=True)

def make_dir(directory):
    os.makedirs(directory, exist_ok=True)

def read_file(path):
    with open(path, 'r', encoding='utf-8') as file:
        content = file.read()
        return content

def get_config():
    if not os.path.exists(config_path) or os.path.getsize(config_path) == 0:
        with open(config_path, 'w', encoding='utf-8') as file:
            json.dump({'address1': '', 'address2': '', 'address3': '', 'address3Type':'', 'comfyuiNodeApi':''}, file)
    try:
        with open(config_path, 'r', encoding='utf-8') as file:
            return json.load(file)
    except Exception as e:
        raise
